### PR TITLE
ACC-45 : Allow to deploy standalone ES

### PR DESCRIPTION
**Pull request by vsellier**
* Allow to deploy an elasticsearch container per instance.
* add a line on the instance's information popup :
screen shot 2017-12-11 at 15 25 03

The configuration is controlled via this new environment variables : 
*  DEPLOYMENT_ES_EMBEDDED
*  DEPLOYMENT_ES_IMAGE
*  DEPLOYMENT_ES_IMAGE_VERSION
*  DEPLOYMENT_ES_HEAP

### CODE REVIEW

File: _functions_es.sh [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+  local wait_time=1
+  local RET=-1
+
+  local temp_file="/tmp/${DEPLOYMENT_ES_CONTAINER_NAME}_${DEPLOYMENT_ES_HTTP_PORT}.txt"
+
+  while [ $count -lt $try -a $RET -ne 0 ]; do
+    count=$(( $count + 1 ))
+    set +e
+
+    curl -s -q http://localhost:${DEPLOYMENT_ES_HTTP_PORT} > ${temp_file}
```
Comment (fdrouet) [inline _functions_es.sh]: No timeout specified here ? what is the default curl timeout ?

File: _functions.sh [hunk 2]
```diff
@@ -148,6 +149,12 @@ Environment Variables
 
   DEPLOYMENT_EXTENSIONS             : Comma separated list of PLF extensions to install. "all" to install all extensions available. Empty string for none. (default: all)
 
+  DEPLOYMENT_ES_ENABLED             : Do we need to configure elasticsearch (default: true; values : true|false)
+  DEPLOYMENT_ES_EMBEDDED            : Do you want to have use an embedded elasticsearch deployment or not (default: true; values: true|false)
```
[the remark this answers is not in the document]
Comment (vsellier) [reply _functions.sh]: log updated

File: _functions.sh [hunk 3]
```diff
@@ -148,6 +149,12 @@ Environment Variables
...
+  DEPLOYMENT_ES_IMAGE               : Which docker image you want to use for standalone elasticsearch (default: exoplatform/elasticsearch)
```
[the remark this answers is not in the document]
Comment (vsellier) [reply _functions.sh]: log updated

File: _functions.sh [hunk 4]
```diff
@@ -148,6 +149,12 @@ Environment Variables
...
+  DEPLOYMENT_ES_IMAGE_VERSION       : which version of the ES image you want to use (default 1.1.0)
```
[the remark this answers is not in the document]
Comment (vsellier) [reply _functions.sh]: log updated

File: _functions.sh [hunk 5]
```diff
@@ -148,6 +149,12 @@ Environment Variables
...
+  DEPLOYMENT_ES_HEAP                : Size of the heap to configure (default: 512m)
```
[the remark this answers is not in the document]
Comment (vsellier) [reply _functions.sh]: log updated

File: _functions_es.sh [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+}
+
+#
+# Drops all Elasticsearch datas used by the instance.
+#
+do_drop_es_data() {
+  echo_info "Dropping es data ..."
+  if ${DEPLOYMENT_ES_ENABLED}; then
+    if ${DEPLOYMENT_ES_EMBEDDED}; then
+      echo_info "Drops ES embedded instance datas ..."
```
[the remark this answers is not in the document]
Comment (vsellier) [reply _functions_es.sh]: log updated

File: _functions_es.sh [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+      rm -rf ${DEPLOYMENT_DIR}/${DEPLOYMENT_ES_PATH_DATA}
+      echo_info "Done."
+    else
+      echo_info "Drops ES docker volume (${DEPLOYMENT_ES_CONTAINER_NAME})..."
```
[the remark this answers is not in the document]
Comment (vsellier) [reply _functions_es.sh]: log improved to trace the 2 actions separatly

File: _functions_es.sh (hunk 1 above)
Comment (vsellier) [reply _functions_es.sh]: I didn't found the default curl timeout value so I specified it.

**Comment by fdrouet** [comment]
LGTM

**Pull request merged**